crypto/attachment.go

```diff
@@ -13,6 +15,178 @@ import (
 	"github.com/pkg/errors"
 )
 
+// AttachmentProcessor2 keeps track of the progress of encrypting an attachment
+// (optimized for encrypting large files).
+// With this processor, the caller has to first allocate
+// a buffer large enough to hold the whole data packet.
+type AttachmentProcessor2 struct {
```

Comment:
```suggestion
type ManualAttachmentProcessor struct {
```

Or something that recalls the manual/aggressive memory management :)

I'd also put the new types at the beginning of the file or in a separate file, followed then by the methods as we do in the rest of the lib

Reply:
I think a separate file would make sense.
would `attachment_manual.go` work with you ?

Reply:
Yup, good for me :+1: